Controllers/Config/InvalidModelStateResponseFactory.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using WebAPI.API.Extensions;
+using WebAPI.Resources;
+
+namespace WebAPI.Controllers.Config
+{
+    public static class InvalidModelStateResponseFactory
+    {
+        public static IActionResult ProduceErrorResponse(ActionContext context)
+        {
+            var errors = context.ModelState.GetErrorMessages();
+            var response = new ErrorResource(errors);
+            return new BadRequestObjectResult(response);
+        }
+    }
+}
```

Comment:
This new factory allows us to tell the pipeline how to return model state errors.  In this case as an ErrorResource object.    The default would have returned: BadRequestObjectResult(actionContext.ModelState)